### ISSUE TITLE
Screen reader doesn't tell how many followers/following someone has (mobile app)

### ISSUE THREAD

**Issue by hs4man21** [bluesky-social/social-app#697]
**Describe the bug**
As demonstrated in the video below (turn audio on), the accessibilityLabels for number of followers and following on profile pages does not include how many followers/following the person has. So this information is currently inaccessible.

**To Reproduce**

1. Use an iPhone (or Android with TalkBack screen reader turned on, but my instructions assume iPhone).
2. Turn on [the VoiceOver screen reader](https://build.washingtonpost.com/resources/accessibility/screen-readers#iOS) in your iPhone's accessibility settings.
3. Open the Bluesky app.
4. Open your profile, and click the followers and following numbers to hear the labels read aloud. Note they're lacking numbers.

**Expected behavior**

The accessibilityLabels for each of these buttons can just be what is shown visually (maybe no `accessibilityLabel` is needed). The `accessibilityHint` should say "Opens following/follower" list for users who aren't sure what the buttons do, but based on my testing with Twitter this should be enough labeling. And it preserves the info shown visually, which is being overwritten by the current labels.

**Screenshots**

https://github.com/bluesky-social/social-app/assets/38192823/e3ffae45-5a0f-4b15-bafc-07020e0c98c1

**Details**

- Platform: iPhone XR, iOS 16.4.1, using VoiceOver screen reader
- Platform version: MacOS: Ventura 13.1, Chrome: 112.0.5615.137
- App version: 1.28
